src/shared/Microsoft.Git.CredentialManager/Settings.cs

```diff
@@ -358,6 +397,30 @@ bool TryGetUriSetting(string envarName, string section, string property, out Uri
                 return false;
             }
 
+            ProxyConfiguration CreateConfiguration(Uri uri, bool isLegacy = false)
+            {
+                // Strip the userinfo, query, and fragment parts of the Uri retaining only the scheme, host, port, and path
+                Uri address = new UriBuilder(uri)
+                {
+                    UserName = string.Empty,
+                    Password = string.Empty,
+                    Query    = string.Empty,
+                    Fragment = string.Empty,
+                }.Uri;
+
+                // Extract the username and password from the URI if present
+                uri.TryGetUserInfo(out string userName, out string password);
+
+                // Get the proxy bypass host names
+                var bypassHosts = new List<string>();
+                if (_environment.Variables.TryGetValue(KnownEnvars.CurlNoProxy, out string noProxyStr))
+                {
+                    bypassHosts = noProxyStr.Split(',').ToList();
```

Comment:
Curl actually _also_ allows separating by spaces: https://github.com/curl/curl/blob/313faa1bcbd2bcc957e660d98360a82c15fe7c07/lib/url.c#L2181. Do we want to do the same?

Reply:
@dscho Is it mutually exclusive or split on either?

```text
foo.com,bar.com example.com

-->

foo.com
bar.com
example.com
```

Reply:
As far as I read the code, it's either. I.e. something like `NO_PROXY.split(/[ ,]+/)`.

Reply:
Added!